src/Core/src/Platform/Windows/KeyboardAcceleratorExtensions.cs

```diff
@@ -0,0 +1,121 @@
+﻿using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Input;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Windows.System;
+
+namespace Microsoft.Maui.Platform
+{
+	public static class KeyboardAcceleratorExtensions
+	{
+		public static void UpdateAccelerator(this MenuFlyoutItemBase platformView, IMenuFlyoutItem menuFlyoutItem)
+		{
+			var keyboardAccelerators = menuFlyoutItem.Accelerators?.ToPlatform();
+
+			if (keyboardAccelerators is null)
+				return;
+
+			foreach (var keyboardAccelerator in keyboardAccelerators)
+				platformView.KeyboardAccelerators.Add(keyboardAccelerator);
+		}
+
+		public static IList<KeyboardAccelerator>? ToPlatform(this IReadOnlyList<IAccelerator> accelerators)
+		{
+			if (accelerators is null)
+				return null;
+
+			List<KeyboardAccelerator> result = new List<KeyboardAccelerator>();
+
+			foreach (var accelerator in accelerators)
+			{
+				var keyboardAccelerators = accelerator.ToPlatform();
+
+				if (keyboardAccelerators is not null)
+				{
+					foreach (var keyboardAccelerator in keyboardAccelerators)
+						result.Add(keyboardAccelerator);
+				}
+			}
+
+			return result;
+		}
+
+		// Single key (A, Delete, F2, Spacebar, Esc, Multimedia Key) accelerators and multi-key
+		// accelerators (Ctrl+Shift+M) are supported.
+		// Gamepad virtual keys are not supported.
+		public static IList<KeyboardAccelerator>? ToPlatform(this IAccelerator accelerator)
+		{
+			if (accelerator is null)
+				return null;
+
+			List<KeyboardAccelerator> result = new List<KeyboardAccelerator>();
+
+			var key = accelerator.Key;
+			var modifiers = accelerator.Modifiers;
+
+			var keyboardAccelerator = new KeyboardAccelerator();
+			keyboardAccelerator.Key = key.ToVirtualKey();
+			if (modifiers is not null)
+			{
+				foreach (var mod in modifiers)
+				{
+					keyboardAccelerator.Modifiers |= mod.ToVirtualKeyModifiers();
+				}
+			}
+			result.Add(keyboardAccelerator);
+
+			return result;
+		}
+
+		internal static VirtualKeyModifiers ToVirtualKeyModifiers(this string modifierMask)
+		{
+			switch (modifierMask.ToLowerInvariant())
+			{
+				case "ctrl":
+					return VirtualKeyModifiers.Control;
+				case "alt":
+					return VirtualKeyModifiers.Menu;
+				case "shift":
+					return VirtualKeyModifiers.Shift;
+				case "win":
+					return VirtualKeyModifiers.Windows;
+				default:
+					return VirtualKeyModifiers.None;
+			}
+		}
+
+		internal static VirtualKey ToVirtualKey(this string key)
+		{
+			if (int.TryParse(key, out int numberKey))
+			{
+				if (numberKey >= 0 && numberKey <= 9)
+				{
+					key = $"Number{key}";
+				}
+			}
```

Comment:
What happens if the user made a mistake and the key is a int but out of the range?  If we do not care, lets merge the ifs with an &&

Reply:
Like Ctrl+11? that is not really a key but multiple keys.

Reply:
Exactly, so the setup is wrong, yet we are not throwing an exception or anything, meaning that we will have an issue at run time (key is revamped maybe?). If the number is out of range, we should throw and out of range exception.